packages/core/src/provider.ts

```diff
@@ -12,4 +12,16 @@ export class SphinxJsonRpcProvider extends JsonRpcProvider {
       cacheTimeout: -1,
     })
   }
+
+  // On OKT Chain, they appear to be running an out of date geth client which is not compatible with the current spec.
+  // For shame!
+  // See this issue for more info: https://github.com/NomicFoundation/hardhat/issues/4010
+  // So on OKT Chain we have to call eth_estimateGas ourselves to handle this....
+  estimateGas(_tx: TransactionRequest): Promise<bigint> {
+    if (_tx.chainId === 66) {
+      return this.send('eth_estimateGas', [_tx])
+    }
+
+    return super.estimateGas(_tx)
+  }
```

Comment:
See comment; it's necessary to override `estimateGas` on OKT Chain due to the fact that they use an outdated geth client. This workaround resolves the issue, but it means we always have to use `SphinxJsonRPCProvider` to interact with that network. I think that is acceptable as we already are very good about using the custom provider. After I implemented this workaround, everything worked properly on the network.